flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -119,7 +120,7 @@
 
 	private boolean shutdown;
 
-	public TaskManagerRunner(Configuration configuration, ResourceID resourceId) throws Exception {
+	public TaskManagerRunner(Configuration configuration, ResourceID resourceId, PluginManager pluginManager) throws Exception {
```

Comment:
In the PRs current state `pluginManager` should be annotated with `Nullable`

Reply:
I am  not sure what should be the expected contract here. ReporterSetup will currently only work with the null pluginManager if no reporters are configured (namedReporters.isEmpty()). We could maybe add a checkNonNull in  the ReporterSetup after the first return due to empty namedReporters.

(TaskManagerRunnerTest is now fixed to initialize the PluginManager properly)